src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.model.Circle;
+import core.basesyntax.model.Figure;
+import core.basesyntax.model.IsoscelesTrapezoid;
+import core.basesyntax.model.Rectangle;
+import core.basesyntax.model.RightTriangle;
+import core.basesyntax.model.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureIndex = new Random().nextInt(4);
+        switch (figureIndex) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), getRandomInt());
+            case 1:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        getRandomDouble(),
+                        getRandomDouble(),
+                        getRandomDouble());
+            case 2:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        getRandomDouble(),
+                        getRandomDouble());
+            case 3:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        getRandomInt(),
+                        getRandomInt());
+            case 4:
+                return new Square(colorSupplier.getRandomColor(), getRandomInt());
+            default:
+                return new Circle(colorSupplier.getRandomColor(), getRandomInt());
```

Comment:
create for each figure a separate method that will be returning a random figure, for example, getRandomSquare() and so on